packages/contracts-bedrock/deploy/019-L2OutputOracleInit.ts

```diff
@@ -0,0 +1,51 @@
+import assert from 'assert'
+
+import { ethers } from 'ethers'
+import { DeployFunction } from 'hardhat-deploy/dist/types'
+import { awaitCondition } from '@eth-optimism/core-utils'
+import '@eth-optimism/hardhat-deploy-config'
+import 'hardhat-deploy'
+
+import {
+  getDeploymentAddress,
+  doOwnershipTransfer,
+} from '../src/deploy-utils'
+
+const deployFn: DeployFunction = async (hre) => {
+  const proxyAdmin = await getDeploymentAddress(hre, 'ProxyAdmin')
+  const l2OutputOracle = await getDeploymentAddress(hre, "L2OutputOracle")
+  const l2OutputOracleProxy = await getDeploymentAddress(hre, 'L2OutputOracleProxy')
+
+  await proxyAdmin.upgradeAndCall(
+    l2OutputOracleProxy,
+    l2OutputOracle,
+    // TODO: how to abi encode here
```

Comment:
https://github.com/ethers-io/ethers.js/blob/master/packages/abi/src.ts/interface.ts#L371

There is a way to access this on the `contract` object

Reply:
`contract.interface.encodeFunctionData("name", [])`

i think

Reply:
https://docs.ethers.org/v5/api/utils/abi/interface/